docker/generate_makefile.py

```diff
@@ -14,8 +14,6 @@
 """Simple generator for local Makefile rules."""
```

Comment:
I'm pretty sure generate_makefile was only introduced for variants so in theory we can get rid of it.
Since we plan on putting more build logic in python we could reuse some of this. But frankly, I think starting fresh is a better idea.

Reply:
Yes i would let @zchcai do this as part of redesign.